packages/resolver/src/resolvers/npm_resolver.ts

```diff
@@ -0,0 +1,34 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { ContractSource } from '../types';
+
+import { Resolver } from './resolver';
+
+export class NPMResolver extends Resolver {
+    private _packagePath: string;
+    constructor(packagePath: string) {
+        super();
+        this._packagePath = packagePath;
+    }
+    public resolveIfExists(importPath: string): ContractSource | undefined {
+        if (!importPath.startsWith('/')) {
+            const [packageName, ...other] = importPath.split('/');
+            const pathWithinPackage = path.join(...other);
+            let currentPath = this._packagePath;
+            const ROOT_PATH = '/';
+            while (currentPath !== ROOT_PATH) {
+                const lookupPath = path.join(currentPath, 'node_modules', packageName, pathWithinPackage);
+                if (fs.existsSync(lookupPath)) {
```

Comment:
Are false positives possible here? Ie, could more than one `lookupPath` resolve in the range (`/`..`this._packagePath`]?

Reply:
Not false positives, but you could have a situation when monorepo used open-zeppelin v1.0 but your package uses open-zeppelin v2.0. So you'll have `./node_modules/open-zeppelin/Ownable.sol` and `../../node_modules/open-zeppelin/Ownable.sol`. This will resolve to the first one. It goes top-down as yarn and npm does.